### ISSUE TITLE
[Bug]: word_timestamps hangs forever when the deploy config has async_chunk: true (the bundled qwen3_tts.yaml default)

### ISSUE THREAD

**Issue by farzanehnakhaee70** [vllm-project/vllm-omni#7505]
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Image: vllm/vllm-omni:v0.28.0
Model: Qwen/Qwen3-TTS-12Hz-1.7B-CustomVoice
Aligner: Qwen/Qwen3-ForcedAligner-0.6B
GPU: TODO — e.g. 1x A100 80GB
Deploy config: the bundled vllm_omni/deploy/qwen3_tts.yaml (2 stages, async_chunk: true, SharedMemoryConnector with codec_streaming: true)
```

</details>

### 🐛 Describe the bug

Describe the bug

With a forced-aligner stage injected, any request carrying "word_timestamps": true never completes and never errors. The connection hangs until the client's own timeout. The server stays healthy throughout (GET /v1/models keeps returning 200), and other requests without word_timestamps are served normally.

Setting async_chunk: false and codec_streaming: false in the deploy config makes it work correctly — audio plus a populated X-Word-Timestamps header. That is currently the only workaround I've found.

This means the documented happy path is broken: the docs recommend --forced-aligner together with --deploy-config vllm_omni/deploy/qwen3_tts.yaml, and that file ships async_chunk: true.

Steps to reproduce
vllm-omni serve Qwen/Qwen3-TTS-12Hz-1.7B-CustomVoice \
  --omni \
  --deploy-config vllm_omni/deploy/qwen3_tts.yaml \
  --trust-remote-code \
  --forced-aligner Qwen/Qwen3-ForcedAligner-0.6B \
  --port 8000
# hangs indefinitely, no response, no error
curl -X POST http://localhost:8000/v1/audio/speech \
  -H 'Content-Type: application/json' \
  -d '{"model": "Qwen/Qwen3-TTS-12Hz-1.7B-CustomVoice",
       "input": "Guten Tag, wie geht es Ihnen?",
       "voice": "vivian",
       "language": "de",
       "response_format": "wav",
       "word_timestamps": true}'

Controls:

Same request without word_timestamps → returns audio normally.
Same request with async_chunk: false + codec_streaming: false in the deploy config → returns audio and X-Word-Timestamps.
Server log

The request is accepted, the aligner stage initializes, stage 0 starts sampling — then nothing. No completion, no error, no code2wav2aligner: empty waveform warning.

(APIServer pid=1) INFO [serving_speech.py:3070] TTS speech request speech-b098cf56ae6a18c1: model=CustomVoice
(APIServer pid=1) INFO [serving_speech.py:3095] Applied extra_params: {'stop_token_ids': [...], 'temperature': 0.1, 'top_p': 1.0, 'repetition_penalty': 1.05}
(StageEngineCoreProc_stage2_replica0 pid=1844) INFO [core.py:413] Resolved pooling config: pooling_type=ALL(source=model_default), seq_pooling_type=LAST(source=model_default), use_activation=True(source=pooler_default), supported_tasks=('token_classify',)
(StageEngineCoreProc_stage0_replica0 pid=466) INFO [configuration_qwen3_tts.py:486] speaker_encoder_config is None. Initializing talker model with default values
(StageEngineCoreProc_stage0_replica0 pid=466) INFO [configuration_qwen3_tts.py:437] code_predictor_config is None. Initializing code_predictor model with default values
(StageEngineCoreProc_stage0_replica0 pid=466) WARNING [sampling_utils.py:57] min_tokens: dropped stop token ids [151645] that exceed the logits vocabulary (3072); the model head cannot emit them.
(APIServer pid=1) INFO:     127.0.0.1:33026 - "GET /v1/models HTTP/1.1" 200 OK      <-- healthcheck, 27s later; server alive
<no further output; request never completes>

Note stage 2 resolves its pooling config fine, so the aligner model itself loads correctly — this is not an OOM or a model-loading failure.

Suspected mechanism

I traced this through the v0.28.0 source; posting it in case it saves triage time, though I haven't confirmed it with a debugger.

On the non-streaming path, word_timestamps extends the requested modalities so the orchestrator drives the request through the aligner stage — serving_speech.py:
output_modalities = ["audio"]
if getattr(request, "word_timestamps", False):
    output_modalities.append(TIMESTAMPS_MODALITY)
The collection loop async for res in generator: only terminates once every requested modality has been produced.
code2wav2aligner (vllm_omni/model_executor/stage_input_processors/forced_aligner.py) skips any source that is not yet complete, and returns an empty list if that is true for all of them:
for i, src in enumerate(source_outputs):
    if not getattr(src, "finished", False):
        # Sentence-final alignment: only run once the audio is complete.
        continue

With async_chunk: true + codec_streaming: true, Code2Wav emits audio incrementally through the SharedMemoryConnector, so it appears the aligner never sees a finished source carrying the complete waveform. Empty input → the aligner stage produces no output → the timestamps modality is never emitted → the generator never finishes. There is no server-side timeout on that path, which explains the silent hang rather than an error.

Expected behaviour

Either:

The aligner receives the completed utterance and returns timestamps under async_chunk: true (preferred — it's the shipped default), or
the combination is rejected at startup or per-request with a clear error, the way the WebSocket path already rejects unmet preconditions, instead of hanging forever.

At minimum, a hang with no diagnostic is the worst outcome — any failure mode with a message would be an improvement.

Possibly related
[vllm-project/vllm-omni#6898](https://github.com/vllm-project/vllm-omni/pull/6898) (Gate async-chunk Code2Wav on non_streaming_mode) touches the same async-chunk / non-streaming interaction.
#5099 (Disable Async Chunk for Unsupported Pipelines) proposes exactly the "disable + warn instead of failing confusingly" behaviour that would cover option 2 above.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.
